doc/Architecture.md

```diff
@@ -0,0 +1,53 @@
+# Architecture
+
+This section describes the overall architecture of Theia. 
+Theia is designed to work as a native desktop application as well as in the context of a browser and a remote server. To support both situations with a single source, Theia runs in two separate processes. Those processes are called _frontend_ and _backend_ respectively, and they communicate through JSON-RPC messages over WebSockets or REST APIs over HTTP. In the case of Electron, the backend, as well as the frontend, run locally, while in a remote context the backend would run on a remote host.
+
+Both the frontend and backend processes have their dependency injection container (see below) to which extensions can contribute.
+
+## Frontend
+
+The frontend process represents the client and renders the UI. In the browser, it simply runs in the rendering loop, while in Electron it runs in an Electron Window, which basically is a browser with additional Electron and node.js APIs. Therefore, frontend code may only access sources that depend on any API that can run in browsers.
+
+The startup of the frontend process will first load the modules of all contributing extensions before it obtains an instance of `FrontendApplication` from the resulting Dependency Injection (DI) container and call `start()` on it.
+
```

Comment:
That may be diving in DI very quickly, having it's own section inside the DI section might be an idea ?

LIke DI -> Application composition -> Frontend / Backend etc..